robot/impl/resource_manager.go

```diff
@@ -179,6 +179,15 @@ func (manager *resourceManager) updateRemoteResourceNames(
 ) bool {
 	manager.logger.CDebugw(ctx, "updating remote resource names", "remote", remoteName)
 	activeResourceNames := map[resource.Name]bool{}
+	// if the robot supports the Connected method then we know that it is for a resource
+	// that might be disconnected. If it is disconnected then we won't update our
+	// resources for that robot as they are in an indeterminant state
+	connectable, ok := rr.(interface{ Connected() bool })
```

Comment:
temp for testing until https://viam.atlassian.net/browse/RSDK-6433 is implemented